core/src/main/java/com/yahoo/oak/BlocksPool.java

```diff
@@ -76,9 +89,25 @@ static void setBlockSize(int blockSize) {
         }
     }
 
+    /**
+     * Sets the preferred block size. This only has an effect if the block pool was never used before.
+     * @param preferredBlockSize the preferred block size
+     */
+    static void preferBlockSize(long preferredBlockSize) {
```

Comment:
Good idea. I'll do that.

Reply:
> What do you think about setting PRE_ALLOC_SIZE_BYTES same way?

Currently, once `preferBlockSize()` is used, we instantiate a new `BlocksPool`.
If we want to add additional user preferences, we would have to defer the instantiation of the pool for later.
This will have a completely different design that I prefer not to get into right now.
Especially since we might refactor this anyway as part of the memory-manager package effort.

Reply:
OK